### PR TITLE
ADM-372 [backend] Generate board data

### PR DESCRIPTION
**Pull request by fengxinhou**
## Summary

Backend generation of board CSV file

## After

Screen Shot 2023-06-20 at 4 58 59 PM

### CODE REVIEW

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 1]
```diff
@@ -605,4 +577,119 @@ private CardCustomFieldKey covertCustomFieldKey(List<TargetField> model) {
...
+			jql = "";
+		}
+		else {
+			jql = "status not in ('" + String.join("','", status) + "')";
+		}
+		return getCardList(baseUrl, boardRequestParam, jql, "nonDone");
+	}
+
+	private List<JiraCard> getCardList(URI baseUrl, BoardRequestParam boardRequestParam, String jql, String cardType) {
+		// TODO 变量名称，log
```
Comment (leeranzhi) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:652]: what's mean for this note?

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 2]
```diff
@@ -605,4 +577,119 @@ private CardCustomFieldKey covertCustomFieldKey(List<TargetField> model) {
...
+		log.info("Start to get first-page {} card information form kanban", cardType);
+		String allCardResponse = jiraFeignClient.getAllDoneCards(baseUrl, boardRequestParam.getBoardId(), QUERY_COUNT,
+				0, jql, boardRequestParam.getToken());
+		if (allCardResponse.isEmpty()) {
+			return Collections.emptyList();
+		}
+		log.info("Successfully get first-page {} card information form kanban", cardType);
```
Comment (leeranzhi) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:659]: Please follow the log specification

File: backend/src/main/java/heartbeat/service/report/GenerateReporterService.java [hunk 3]
```diff
@@ -102,6 +130,70 @@ public static Map<String, String> getRepoMap(CodebaseSetting codebaseSetting) {
...
+		}
+
+		if (isArray) {
+			result = "[0]" + result;
+		}
+
+		return result;
+	}
+
 	public synchronized ReportResponse generateReporter(GenerateReportRequest request) {
```
Comment (leeranzhi) [inline backend/src/main/java/heartbeat/service/report/GenerateReporterService.java:197]: why add `synchronized` for this fuction?

**Review by leeranzhi — CHANGES_REQUESTED** [review]

File: backend/src/main/java/heartbeat/service/report/GenerateReporterService.java (hunk 3 above)
Comment (TingLiMio) [reply backend/src/main/java/heartbeat/service/report/GenerateReporterService.java:197]: This `synchronized` was added before, not in this PR. 
The reason is front end will send two report requests at the same time.

**Review by leeranzhi — APPROVED** [review]

**Review by wangwang-zhang — APPROVED** [review]

**Pull request merged**